src/util/prolog/prologResult.ts

```diff
@@ -9,22 +9,32 @@ const booleanAnswerRegExp = /^(true|false)( ;|,|.)$/;
 // Second group contains the variable names
 const variableNameRegExp = /(^| )([A-Z]\w*)/g;
 
+/** Handles the results for a prolog query returned by PrologSession.executeQuery() */
 export class PrologResult {
 	private readonly rawResults: string[];
 	private results: Map<string, Array<string | boolean>> | undefined;
 
+	/**
+		* Simply forwards the raw results to its respective attribute
+		* @param {Array<string>} rawResults - The string array containing the results
+		*/
 	constructor(rawResults: string[]) {
 		this.rawResults = rawResults;
 	}
 
-	public getRawResults() {
+	/**
+		* rawResults getter-Methode
+		* @return {Array<string>} The raw results
+		*/
+	public getRawResults(): string[] {
```

Comment:
Either make the return type readonly (either by writing `readonly string[]` or `ReadonlyArray<string>`) or return a copy.